services/data-info/src/data_info/services/metadata.clj

```diff
@@ -36,13 +37,170 @@
   [avu]
   (re-find ipc-regex (:attr avu)))
 
+(defn- authorized-avus
+  "Validation to make sure the AVUs aren't system AVUs. Throws a slingshot error
+   map if the validation fails."
+  [avus]
+  (when (some ipc-avu? avus)
+    (throw+ {:error_code ERR_NOT_AUTHORIZED
+             :avus avus})))
+
+(defn- get-readable-path
+  [cm user data-id]
+  (let [path (:path (uuids/path-for-uuid cm user data-id))]
+    (validators/path-readable cm user path)
+    path))
+
 (defn- list-path-metadata
   "Returns the metadata for a path. Passes all AVUs to (fix-unit).
    AVUs with a unit matching IPCSYSTEM are filtered out."
-  [cm path]
-  (remove
-   ipc-avu?
-   (map fix-unit (get-metadata cm (ft/rm-last-slash path)))))
+  [cm path & {:keys [system] :or {system false}}]
+  (let [fixed-metadata (map fix-unit (get-metadata cm (ft/rm-last-slash path)))]
+  (if system
+      fixed-metadata
+      (remove ipc-avu? fixed-metadata))))
+
+(defn- reserved-unit
+  "Turns a blank unit into a reserved unit."
+  [avu-map]
+  (if (string/blank? (:unit avu-map))
+    paths/IPCRESERVED
+    (:unit avu-map)))
+
+(defn metadata-get
+  "Returns the metadata for a path. Filters out system AVUs
+   if :system true is not passed to it, and replaces
+   units set to ipc-reserved with an empty string."
+  [user data-id & {:keys [system] :or {system false}}]
+  (with-jargon (cfg/jargon-cfg) [cm]
+    (validators/user-exists cm user)
+    (let [path (get-readable-path cm user data-id)]
+      {:irods-avus (list-path-metadata cm path :system system)
+       :path path})))
```

Comment:
Most of this file is copied from the terrain version. Note however that this endpoint returns the path, since upstream endpoints wanted to return it to the UI.